lib/engine/game/g_18_mag/game.rb

```diff
@@ -616,31 +620,47 @@ def check_distance(route, visits)
           raise GameError, 'Must visit minimum of two non-mine stops' if visits.sum(&:visit_cost) < 2
         end
 
+        def compute_stops(route, train = nil)
+          train ||= route.train
+          distance = if gc_train?(route) && !other_gc_train?(route)
+                       gc_train_distance(route.train.distance)
+                     else
+                       route.train.distance
+                     end
+          train_dup = train.dup
+          train_dup.distance = distance
+
```

Comment:
When the train has been converted to +1 using the supporter, stops breaks when also adding gc (for example 2+1 turns into 2+3), the base compute_stops only returns two cities/offboards and 1 town. 
Duplicating the train and setting the correct distance fixes the issue.

Would a better alternative be to modify compute_stops to take distance=nil and pass the correct distance?
maybe copy the entire base method here and modify accordingly?

Reply:
No need to do that, but I think this method can be cleaned up while keeping the same logic--I'll post my suggested code.